src/gnark_backend_glue/mod.rs

```diff
@@ -64,6 +65,18 @@ struct KeyPair {
     verifying_key: *const c_char,
 }
 
+#[derive(Error, Debug)]
+pub enum GnarkBackendError {
+    #[error("an error occured while serializing the circuit")]
+    SerializeCircuitError,
+    #[error("an error occured while serializing a key")]
+    SerializeKeyError,
+    #[error("an error occured while deserializing a proof")]
+    DeserializeProofError,
```

Comment:
We could do something like this in order to preserve the original error message:
```suggestion
    #[error("an error occured while deserializing a proof: {}")]
    DeserializeProofError(String),
```

Then the usage would be:
```rust
let bytes = c_str
    .to_str()
    .map_err(|e| GnarkBackendError::DeserializeProofError(e.to_string())?
    .as_bytes();
```